### PR TITLE
Login plugin to store user name in a cookie

### PR DESCRIPTION
**Pull request by koke24**
Hi!

Here is a plugin to store user name in a cookie in the browser. It is almost independent but It needs a minimal change in tiddlyweb plugin to allow it to retrieve user name from the stored cookie and use it as the user name for the TiddlyWiki.

Best regards

### CODE REVIEW

File: plugins/tiddlywiki/login/files/login.tid [hunk 1]
```diff
@@ -0,0 +1,25 @@
+created: 20160115112840267
+modified: 20160118125350050
+revision: 0
+tags: 
+title: Login
+type: text/vnd.tiddlywiki
+
+<$reveal type="match" state="$:/status/UserName" text="">
+No he encontrado tu usuario. Por favor, indica tu nombre o nick para identificar tus entradas.
```
Comment (felixhayashi) [inline plugins/tiddlywiki/login/files/login.tid]: Amigo, yo no hablo español :)

File: plugins/tiddlywiki/login/login.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  self.parse();
+}
+Cookies.prototype.parse = function() {
+  var self = this;
+  self.__parsed__ = true;
+  var cookies = document.cookie.split(/\s*;\s*/g);
+  console.log("cookies: "+document.cookie);
+  var cookie;
+  for(var c = 0; c<cookies.length; c++) {
+    console.log("cookie: "+cookies[c]);
```
Comment (felixhayashi) [inline plugins/tiddlywiki/login/login.js]: probably remove console.logs

File: plugins/tiddlywiki/login/login.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    cookie = cookies[c].split("=");
+    self[cookie[0]] = cookie[1];
+    //if (cookie[0]=='expires') self.__persist__ = true;
```
Comment (felixhayashi) [inline plugins/tiddlywiki/login/login.js]: …and remove comments

File: plugins/tiddlywiki/login/login.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  if (days==0) {
+    str += ";expires=Thu, 01 Jan 1970 00:00:00 GMT";
+    self[name]=null;
+    delete self[name];
+  }
+  document.cookie = str;
+  console.log("setting cookie: "+str);
+}
+
+var logger = new $tw.utils.Logger("Login");
```
Comment (felixhayashi) [inline plugins/tiddlywiki/login/login.js]: why is that needed?

File: plugins/tiddlywiki/login/login.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    for(var a in data) {
+      //console.log("changed: "+a);
+      var matcher = /^\$:\/status\/UserName$/.exec(a);
+      if (matcher) {
+        var cookies = new Cookies();
+        //console.log(""+$tw.wiki.getTiddler(a).getFieldString('text'));
+        var userName = $tw.wiki.getTiddlerText(a);
+        var cookie = $tw.wiki.getTiddlerText(CONFIG_COOKIE_TIDDLER,DEFAULT_COOKIE_TIDDLER);
+        cookies.set(cookie, userName, userName?100:0);
+        if (!userName) $tw.wiki.addTiddler(new $tw.Tiddler("$:/status/IsLoggedIn", { text: "no"}));
```
Comment (felixhayashi) [inline plugins/tiddlywiki/login/login.js]: You should look at existing TW code to see how it is formatted, if I am not mistaken: no spaces between `if (`, always opening brackets after "if"

File: plugins/tiddlywiki/login/plugin.info [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+    "title": "$:/plugins/tiddlywiki/login",
+    "description": "Login plugin",
+    "author": "Koke Laast",
+    "version": "1.0.0",
+    "core-version": ">=5.0.8",
```
Comment (felixhayashi) [inline plugins/tiddlywiki/login/plugin.info]: are you sure with 5.0.8? better choose a newer version e.g. 5.1.5

**Comment by felixhayashi** [comment]
Hi @koke24 I added some remarks. If you recreate your pull request, it would be great if you would "squash your commits" into a single commit. This helps to keep the commit history clean and to make your changes comparable. See also:
  
    * [Why squash git commits for pull requests?]( http://programmers.stackexchange.com/q/263164)
    * [Squashing Github pull requests into a single commit](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit)
    * [How to Rebase a Pull Request](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request)

**Comment by koke24** [comment]
Thank you, @felixhayashi. I will review your remarks on Monday. I apologize for my inexperience.

**Comment by felixhayashi** [comment]
No problem. It's just that chances are better that your code enters the TW5 repository if you did some cleanup before and stick to the code style conventions ;)

**Comment by koke24** [comment]
Well, after a lot of time playing with git I think I have come to a clean repository. If something is wrong, please, don't hesitate to tell me. This is my first github contribution: thank you for your patience.

**Comment by zhilongjia** [comment]
@koke24 It seems the login plugin cannot work soloely. The username was not kept after a chrome browser refresh. A related quesiton in the [TiddlyWiki Google Group](https://groups.google.com/forum/#!topic/tiddlywiki/eM3Lqm1IJKU) Thank you.

**Comment by koke24** [comment]
> @koke24 It seems the login plugin cannot work soloely. The username was not kept after a chrome browser refresh. A related quesiton in the [TiddlyWiki Google Group](https://groups.google.com/forum/#!topic/tiddlywiki/eM3Lqm1IJKU) Thank you.

Thank you for your interest. Unfortunately I can't keep up with this request.
